pkg/provision/storage/commonStorage.go

```diff
@@ -144,6 +154,12 @@ func (p *CommonStorageProvisioner) rewriteContainerVolumeMounts(workspaceId, pvc
 		additionalVolumes[additionalVolume.Name] = true
 	}
 
+	// When we persist the home directory, a volumeMount is created that references the PVC volume.
+	// This volumeMount should not be rewritten
+	if shouldPersistHomeDirectory(workspace) {
+		additionalVolumes[pvcName] = true
+	}
+
```

Comment:
Could you explain what this is doing and why it's necessary? Seems like a steep cost to have to pass in the full `common.DevWorkspaceWithConfig` rather than just a DevWorkspaceTemplateSpec just for this.

This reads like a misuse of the `additionalVolumes` struct -- it will work because we're adding a container that references the PVC volume directly, but in general we assume that containers aren't directly referring to the PVC anyways. I worry that we're breaking the general flow of steps in `ProvisionStorage` by passing data to `rewriteContainerVolumeMounts` that's already half processed and expecting the function to detect and handle that.

The purpose behind `additionalVolumes` is to hold volumes mounted to the workspace container that _aren't_ controlled by DWO's storage provisioners (i.e. are not the per-user/per-workspace PVC) -- it's strange to optionally add the per-user/per-workspace PVC to this list in that context.